### PR TITLE
fjord: Fastlz GasPriceOracle

### PR DESCRIPTION
**Pull request by danyalprout**
**Description**

This PR adds support for improving the estimation of L1 fees by compressing the transaction with fastlz, then using a linear regression to estimate the final Brotli 10 compressed size.

* [Specs](https://github.com/ethereum-optimism/specs/pull/25)
* [Go Implementation](https://github.com/ethereum-optimism/optimism/pull/9618)

**Tests**

* Added unit testing for Solidity implementation
* The Go [PR](https://github.com/ethereum-optimism/optimism/pull/9618) includes fuzz testing for the implementations against a C based fastlz/op-geth cost functions.

### CODE REVIEW

File: bedrock-devnet/devnet/__init__.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def devnet_l2_allocs(paths):
 
     # For the previous forks, and the latest fork (default, thus empty prefix),
     # move the forge-dumps into place as .devnet allocs.
-    for suffix in ["-delta", ""]:
+    for suffix in ["-delta", "-ecotone", ""]:
```
[the remark this answers is not in the document]
Comment (tynes) [reply bedrock-devnet/devnet/__init__.py:173]: This is a good suggestion but we will need to rethink the genesis generation stuff in the future

File: packages/contracts-bedrock/src/L2/GasPriceOracle.sol [hunk 2]
```diff
@@ -51,6 +84,17 @@ contract GasPriceOracle is ISemver {
         isEcotone = true;
     }
 
+    /// @notice Set chain to be Fjord chain (callable by depositor account)
+    function setFjord() external {
+        require(
+            msg.sender == L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).DEPOSITOR_ACCOUNT(),
```
Comment (tynes) [inline packages/contracts-bedrock/src/L2/GasPriceOracle.sol]: While its fine to make the call here, we did turn the `DEPOSITOR_ACCOUNT` into a constant, so I think its fine to inline that constant. You do not need to make this change if you do now want to but if you do then could you also inline the constant inside of `setEcotone` as well?

File: packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol [hunk 3]
```diff
@@ -108,12 +108,20 @@ contract GasPriceOracleBedrock_Test is GasPriceOracle_Test {
         assertEq(success, false);
         assertEq(returndata, hex"");
     }
+
+    /// @dev Tests that Fjord cannot be activated without activating Ecotone
+    function test_setFjord_withoutEcotone_reverts() external {
+        vm.startPrank(depositor);
```
Comment (tynes) [inline packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol]: small nit: no need to start + stop, just regular `prank` should work?

File: packages/contracts-bedrock/src/L2/GasPriceOracle.sol [hunk 4]
```diff
@@ -24,23 +25,55 @@ contract GasPriceOracle is ISemver {
...
 
     /// @notice Semantic version.
-    /// @custom:semver 1.2.0
-    string public constant version = "1.2.0";
+    /// @custom:semver 1.3.0
+    string public constant version = "1.3.0";
+
+    /// @notice This is the intercept value for the linear regression used to estimate the final size of the
+    ///         compressed transaction.
+    int32 private constant COST_INTERCEPT = -42_585_600;
```
Comment (tynes) [inline packages/contracts-bedrock/src/L2/GasPriceOracle.sol:34]: small nit: non standard spacing with newlines

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Just small nits, generally looks good to me

File: packages/contracts-bedrock/src/L2/GasPriceOracle.sol (hunk 2 above)
Comment (danyalprout) [reply packages/contracts-bedrock/src/L2/GasPriceOracle.sol]: I'll do it now, just an FYI it'll change the bytecode, so I'll need to bump the specs PR too.

**Review by sebastianst — APPROVED** [review]
Still LGTM!

**Pull request merged**